src/ui.form-date-picker-range/UDatePickerRange.vue

```diff
@@ -185,205 +33,59 @@ import {
   getYearDateList,
   getQuartersDateList,
   getMonthsDateList,
-} from "./utilDateRange.js";
+  type DatePeriodRange,
+} from "./utilDateRange.ts";
 
-import useAttrs from "./useAttrs.js";
+import useAttrs from "./useAttrs.ts";
 import { useAutoPosition } from "../composables/useAutoPosition.ts";
-import { useLocale } from "./useLocale.js";
-import { useUserFormat } from "./useUserFormat.js";
+import { useLocale } from "./useLocale.ts";
+import { useUserFormat } from "./useUserFormat.ts";
 
-import defaultConfig from "./config.js";
+import defaultConfig from "./config.ts";
 import {
   UDatePickerRange,
   DATE_PICKER_BUTTON_TYPE,
   DATE_PICKER_INPUT_TYPE,
-  PERIOD,
   INPUT_RANGE_FORMAT,
-  SHIFT_ACTION,
-} from "./constants.js";
+  ShiftAction,
+  Period,
+} from "./constants.ts";
+
+import type { Ref, WritableComputedRef } from "vue";
+import type {
+  IsDatePeriodOutOfRange,
+  ShiftActions,
+  SortedLocale,
+  UDatePickerRangeProps,
+  UDatePickerRangeInputsAttrs,
+  UDatePickerRangePeriodMenuAttrs,
+} from "./types.ts";
+import type { RangeDate } from "../ui.form-calendar/types.ts";
+import type { ComponentExposed } from "vue-component-type-helpers";
```

Comment:
There is no reason to add extra dependancy to the project. 
Just use the code from the source: https://www.npmjs.com/package/vue-component-type-helpers?activeTab=code